static/templates/message_edit_form.hbs

```diff
@@ -16,7 +16,7 @@
                       list_placeholder=(t 'Filter streams')}}
                 </div>
                 <i class="fa fa-angle-right" aria-hidden="true" {{#unless is_stream_editable}}style="display:none"{{/unless}}></i>
-                <input type="text" placeholder="{{topic}}" value="{{topic}}" class="message_edit_topic" id="message_edit_topic" autocomplete="off" />
+                <input type="text" placeholder="{{topic}}" value="{{topic}}" class="message_edit_topic" id="message_edit_topic" autocomplete="off" {{#unless should_topic_be_visible}}style="display:none"{{/unless}} />
```

Comment:
I think it should be better is you use `.notvisible` class instead of using `style` inside HTML element. 
https://github.com/zulip/zulip/blob/9094a591e4b8d7f82ae010fbc91de142b66f68ae/static/styles/zulip.css#L550-L558